### ISSUE TITLE
Tabindex doesn't default as the spec says

### ISSUE THREAD

**Issue by chaals** [w3c/html#199]
According to the current spec

> An element with the tabindex attribute specified is interactive content.
> The tabIndex IDL attribute['s] default value is 0 for elements that are focusable and -1 for elements that are not focusable.

Briefly testing blink, Safari and Firefox, elements get 0 or -1 according to their _default_ focusability - i.e apparently from some list of "these kind of elements are focusable, those aren't". Which isn't quite what the spec says.

James Craig @cookiecrook [argued in Bugzilla](https://www.w3.org/Bugs/Public/show_bug.cgi?id=23347) that rather than -1, elements that cannot be focused by default should get `undefined`.

**Comment by chaals** [comment]
A currently [half-done test](http://chaals.github.io/testcases/default-tabindex.html). It really needs better explanations than "use the source".

**Comment by chaals** [comment]
Looking at rough results still, it seems that there are a set of basic focusable elements browsers list. Which seems pretty dumb, but browsers seem consistent - although I haven't yet tested edge/IE.

**Comment by chaals** [comment]
Proposed for 5.1: Note that browsers **must** set the IDL attribute as 0 for elements that are focusable by default, and _should_ set it to 0 for elements that are focusable, and otherwise _should_ set it to -1

**Comment by chaals** [comment]
New proposal. Add an informative warning that _currently_ browsers do something basic, apparently working from a static list of elements. E.g. `<a name="foo" id="foo">nothing interactive here, can't tab to it</a>` gets an IDL `tabindex` of 0. But `<span onclick="handler" accesskey="d">` gets -1.

**Comment by chaals** [comment]
This was closed by #451
